api/geyser/src/main/java/org/geysermc/geyser/api/custom/items/CustomItemData.java

```diff
@@ -27,119 +27,70 @@
-        this.registrationType = registrationType;
-        this.name = name;
-
-        this.displayName = name;
-        this.allowOffhand = false;
-        this.textureSize = 16;
-
-        this.renderOffsets = null;
-    }
-
+public interface CustomItemData {
     /**
-     * Gets the registration types of the item.
+     * Gets the item's name.
      *
-     * @return the registration types of the item.
+     * @return the item's name
      */
-    public CustomItemRegistrationTypes registrationTypes() {
-        return this.registrationType;
-    }
+    @NonNull String name();
 
     /**
-     * Gets the item's name.
+     * Gets the registration types of the item.
      *
-     * @return the item's name
+     * @return the registration types of the item.
      */
-    public @NonNull String name() {
-        return this.name;
-    }
+    CustomItemRegistrationTypes registrationTypes();
 
     /**
      * Gets the item's display name. By default, this is the item's name.
      *
      * @return the item's display name
      */
-    public @NonNull String displayName() {
-        return this.displayName;
-    }
-
-    /**
-     * Sets the item's display name. By default, this is the item's name.
-     *
-     * @param displayName the item's display name
-     */
-    public void displayName(@NonNull String displayName) {
-        this.displayName = displayName;
-    }
+    @NonNull String displayName();
 
     /**
      * Gets if the item is allowed to be put into the offhand.
      *
      * @return true if the item is allowed to be used in the offhand, false otherwise
      */
-    public boolean allowOffhand() {
-        return this.allowOffhand;
-    }
-
-    /**
-     * Sets if the item is allowed to be put into the offhand.
-     *
-     * @param allowOffhand true if the item is allowed to be used in the offhand, false otherwise
-     */
-    public void allowOffhand(boolean allowOffhand) {
-        this.allowOffhand = allowOffhand;
-    }
+    boolean allowOffhand();
 
     /**
      * Gets the item's texture size. This is to resize the item if the texture is not 16x16.
      *
      * @return the item's texture size
      */
-    public int textureSize() {
-        return this.textureSize;
-    }
-
-    /**
-     * Sets the item's texture size. This is to resize the item if the texture is not 16x16.
-     *
-     * @param textureSize the item's texture size
-     */
-    public void textureSize(int textureSize) {
-        this.textureSize = textureSize;
-    }
+    int textureSize();
 
     /**
      * Gets the item's render offsets. If it is null, the item will be rendered normally, with no offsets.
      *
      * @return the item's render offsets
      */
-    public @Nullable CustomRenderOffsets renderOffsets() {
-        return this.renderOffsets;
+    @Nullable CustomRenderOffsets renderOffsets();
+
+    static CustomItemData.Builder builder(@NonNull String name, @NonNull CustomItemRegistrationTypes registrationTypes) {
```

Comment:
Is there any reason these are provided as method parameters rather than being part of the builder?